core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -149,7 +149,12 @@ private[spark] object Utils extends Logging {
 
   /** Deserialize an object using Java serialization and the given ClassLoader */
   def deserialize[T](bytes: Array[Byte], loader: ClassLoader): T = {
-    val bis = new ByteArrayInputStream(bytes)
+    deserialize(ByteBuffer.wrap(bytes), loader)
+  }
+
+  /** Deserialize an object using Java serialization and the given ClassLoader */
+  def deserialize[T](bytes: ByteBuffer, loader: ClassLoader): T = {
+    val bis = new ByteBufferInputStream(bytes)
```

Comment:
I think you misunderstood my comment.  I agree, this change is fine.  Its correctness is clear, and definitely covered by tests.

My initial concern was that this might be some super-common code path, and so even a slight modification may effect performance (should be optimized away by the jit anyway but thought it was worth thinking about).

Anyway, first of all it was a really minor concern, and second of all it turned out to be totally irrelevant since its *not* a commonly used method.  I was just trying to help out any other reviewers from also trying to track this down.  Sorry for the confusion :)